File: src/codeforces/contest1324/ProblemA.java
package codeforces.contest1324;

import java.util.Scanner;

public class ProblemA {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int t = scanner.nextInt();
        for (int i = 0; i < t; i++) {
            int n = scanner.nextInt();
            int counter = 0;
            for (int j = 0; j < n; j++) {
                counter += (scanner.nextInt() & 1);
            }
            if (counter == 0 || counter == n) {
                System.out.println("YES");
            } else {
                System.out.println("NO");
            }
        }
    }
}

File: src/codeforces/contest1710/ProblemA.kt
package codeforces.contest1710

import java.io.BufferedReader
import java.io.InputStream
import java.io.InputStreamReader
import java.util.*

fun main(args: Array<String>) {
    val tests = nextInt()
    for (t in 0 until tests) {
        val n = nextInt()
        val m = nextInt()
        val k = nextInt()
        val arr = IntArray(k) { nextInt() }
        val ans = solve(n, m, arr).or(solve(m, n, arr))
        println(if (ans) "Yes" else "No")
    }
}

private fun solve(n: Int, m: Int, array: IntArray): Boolean {
    var checkH = 0L
    var hasThree = false
    for (a in array) {
        if (n * 2 <= a) checkH += a / n
        if (n * 3 <= a) hasThree = true
    }
    return checkH >= m && (m % 2 == 0 || hasThree)
}


class MyReader(inputStream: InputStream) {
    private val reader = BufferedReader(InputStreamReader(inputStream))
    private var tokenizer: StringTokenizer? = null

    fun next(): String {
        while (true) {
            tokenizer.let {
                if (it == null || !it.hasMoreTokens()) {
                    tokenizer = StringTokenizer(reader.readLine())
                } else {
                    return it.nextToken()
                }
            }
        }
    }
}

val reader = MyReader(System.`in`)
fun next() = reader.next()
fun nextInt() = next().toInt()
fun nextLong() = next().toLong()
fun nextDouble() = next().toDouble()
fun nextStrings(n: Int) = Array(n) { next() }
fun nextInts(n: Int) = IntArray(n) { nextInt() }
fun nextLongs(n: Int) = LongArray(n) { nextLong() }
fun nextDoubles(n: Int) = DoubleArray(n) { nextDouble() }